Resources/public/js/recommendationrestclient.js

```diff
@@ -59,12 +60,23 @@
             }
         };
 
-        var attributes = '';
         for (var i = 0; i < this.attributes.length; i++) {
             attributes = attributes + '&attribute=' + this.attributes[i];
         }
 
-        xmlhttp.open('GET', this.endpointUrl + this.scenario + '.json' + '?numrecs=' + this.limit + '&' + 'contextitems=' + this.contextItems + attributes + '&contenttype=' + this.contentType + '&outputtype=' + this.outputType + '&categorypath=' + encodeURIComponent(this.categoryPath) + '&lang=' + this.language, true);
+        requestQueryString = [
+            this.endpointUrl,
+            this.scenario,
+            '.json?numrecs=', this.limit,
+            '&contextitems=', this.contextItems,
+            '&contenttype=', this.contentType,
+            '&outputtype=', this.outputType,
```

Comment:
I would use `encodeURIComponent` for all parameters (including the *attributes*) just to be sure we produce a valid URI no matter which parameters are used.